app/services/proofing/socure/reason_codes/api_client.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Socure
+    module ReasonCodes
+      class ApiClient
+        class ApiClientError < StandardError; end
+
+        def download_reason_codes
+          http_response = make_reason_code_http_request
+          http_response.body['reasonCodes']
+        rescue Faraday::ConnectionFailed,
+               Faraday::ServerError,
+               Faraday::SSLError,
+               Faraday::TimeoutError,
+               Faraday::ClientError,
+               Faraday::ParsingError => e
```

Comment:
Absolutely not for your PR here, but I feel like our app is all over the place in how we handle different Faraday errors.

For example, where we stop trying to make `fetch` happen with LexisNexis:
https://github.com/18F/identity-idp/blob/main/app/services/doc_auth/lexis_nexis/request.rb#L20-L21

Versus what we handle in the RiscDeliveryJob: https://github.com/18F/identity-idp/blob/main/app/jobs/risc_delivery_job.rb#L6-L14

Mostly rhetorically, are these differences intentional? 🤷‍♂️ I wonder if it would make sense to have an app-wide `FATAL_FARADAY_ERRORS` or something?

This is _way_ out of scope for your PR obviously.

Reply:
We can, though I'm sure that different cases may have different behaviors. That isn't something I want to approach in this change.